From b7468b3ab20268983742e462b225d7e1a8b1c667 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 03 Sep 2021 19:16:17 +0300
Subject: [PATCH] [SW-0] arc: simplify i2c_s enable condition

The condition of ((A == B) == 0) is just a complex way to write
(A != B).
Replace the complicated one with the latter one.

Change-Id: I976cee4548de81af3c7c5ea6ca499a9d6fe4597c
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index b4cffe7..8573e73 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -236,8 +236,8 @@
 			/* if debug I/F is enabled then fall back on I2C_S */
 			ret = preboot_fuse_read(EFUSE_SYSCFG_OFFSET,
 						&config_fuse);
-			if ((!ret)  && ((config_fuse ==
-				EFUSE_SYSCFG_DISABLE_TWI_IF_MASK) == 0)) {
+			if (!ret && config_fuse !=
+					EFUSE_SYSCFG_DISABLE_TWI_IF_MASK) {
 				security_i2c_s_enable();
 			}
 		}
diff --git a/arch/greco/arc_elf32/common.c b/arch/greco/arc_elf32/common.c
index 60be7d2..06b05e8 100644
--- a/arch/greco/arc_elf32/common.c
+++ b/arch/greco/arc_elf32/common.c
@@ -236,8 +236,8 @@
 			/* if debug I/F is enabled then fall back on I2C_S */
 			ret = preboot_fuse_read(EFUSE_SYSCFG_OFFSET,
 						&config_fuse);
-			if ((!ret)  && ((config_fuse ==
-				EFUSE_SYSCFG_DISABLE_TWI_IF_MASK) == 0)) {
+			if (!ret && config_fuse !=
+					EFUSE_SYSCFG_DISABLE_TWI_IF_MASK) {
 				security_i2c_s_enable();
 			}
 		}
